pkg/apis/core/v1alpha1/trigger_types.go

```diff
@@ -23,3 +23,13 @@ type StartOnSpec struct {
 	// UIButtons that can trigger a start/restart.
 	UIButtons []string `json:"uiButtons" protobuf:"bytes,2,rep,name=uiButtons"`
 }
+
+type CancelOnSpec struct {
+	// StartAfter indicates that events before this time should be ignored.
+	//
+	// +optional
+	CancelAfter v1.Time `json:"cancelAfter,omitempty" protobuf:"bytes,1,opt,name=cancelAfter"`
```

Comment:
are you sure we need this? it wasn't necessary for RestartOn, and i thought we rejiggered the StartOn implementation to make this obsolete

Reply:
nope, we don't!